agent/task.go

```diff
@@ -203,6 +210,12 @@ func (tm *taskManager) run(ctx context.Context) {
 			if cancel != nil {
 				// cancel outstanding operation.
 				cancel()
```

Comment:
@jinuxstyle We are going to merge this PR. This should address this issue when encountering in the docker daemon integration. There is still a race condition that needs to be tackled as part of docker/docker#24858 but that is going to be a more complex PR.